---
title: "iRNA_seq_counttable_analysis"
output:
  html_document: default
  pdf_document: default
---

##Analysis for total rna library

the countable is generated by iRNA_seq pipeline (featureCounts), use edgeR to analyze this countable

seems manually edgeR has much fewer significant genes.....

*FeatureCounts output by iRNA-seq:* [use the intron regions= longest isoform - genebank mRNA [to get more accurate intron annotations]  ]18938 genes for exon gtf files; 17285 genes for intron gtf files
1707 genes only in exon file but not intron gtf file: some are indeed single exon genes, but some are quite normal genes 
one_example: ~/Pictures/example_gene.png

```{r echo=F}
library(edgeR)
counttable_path <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_intron_counttable.txt"
counttable_path2 <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_exon_counttable.txt"
intron_table <- read.table(counttable_path,sep="\t",header=T)
exon_table <- read.table(counttable_path2,sep="\t",header=T)
colnames(intron_table)[9:12] <- c('1487_basal',"2139_basal","1487_treat","2139_treat")
yy <- DGEList(counts=intron_table[9:12],genes=intron_table$Symbol,group=factor(c("N","N","T","T")))

colnames(exon_table)[9:12] <- c('1487_basal',"2139_basal","1487_treat","2139_treat")
yy2 <- DGEList(counts=exon_table[9:12],genes=exon_table$Symbol,group=factor(c("N","N","T","T")))

```

*MISTAKES:*
clearly, I wrote this line [o <- order(rowSums(yy2$counts),decreasing = T)] as [o <- order(rowSums(yy$counts),decreasing = T) ] before; 

```{r echo=F}
o <- order(rowSums(yy$counts),decreasing = T)
yy <- yy[o,]
idx <- rowSums(cpm(yy)>2)>=2  # ~10000 genes left
yy <- yy[idx,]
yy$samples$lib.size <- colSums(yy$counts)
yy <- calcNormFactors(yy)


o <- order(rowSums(yy2$counts),decreasing = T) 
yy2 <- yy2[o,]
idx <- rowSums(cpm(yy2)>2)>=2  # ~10000 genes left
yy2 <- yy2[idx,]
yy2$samples$lib.size <- colSums(yy2$counts)
yy2 <- calcNormFactors(yy2)

print("exon reads")
yy2$samples
print("intron reads")
yy$samples
```

##comparisons between total_exon & total_intron:
1.the correlations between samples of total_exon is much higher than total_intron (spearman correlation) [spearman correlation coefficients(~0.84) is lower than pearson(~0.92) for total_intron; haven't checked on total_exon]
2. the clustering of total_intron is: treat vs basal; but for total_exon is: 1487 vs 2139
=> may show that intron-based change is big, however, exon-based change is much lower (short treatment time)


```{r echo=F}
library(corrplot)
plotMDS(yy)
patient <- factor(c("1487","2139","1487","2139"))
tissue <- factor(c("N","N","T","T"))
design <- model.matrix(~patient+tissue)
rownames(design) <- c("1487N","2139N","1487T","2139T")
#design

#estimate teh dispersino
yy <- estimateGLMCommonDisp(yy,design)
yy <- estimateGLMTrendedDisp(yy,design)
yy <- estimateGLMTagwiseDisp(yy,design)

#yy$common.dispersion

plotBCV(yy,cex=0.4)  ##there's some problem with the BCV plot; too few dots....

#corrlation matrix 
M <- cor(data.frame(cpm(yy)),method="spearman")
corrplot.mixed(M,order="hclust")

# home-made PCA; log transform first on library-normalized datasets(many 0s; so try with log(x+1))
transformed.yy <- data.frame(cpm(yy))
transformed.yy <- log(transformed.yy+1)
transformed.yy <- as.data.frame(t(transformed.yy))
transformed.yy.pca <- prcomp(transformed.yy,
                center=T,
                scale.=T)


#print(transformed.yy.pca)
summary(transformed.yy.pca)
plot(transformed.yy.pca, type="l") # the first two pca cannot interpret the whole data (22% and 12%)

y2<- cbind(transformed.yy,treatment=c("1487_basal" ,"2139_basal" ,"1487_treat", "2139_treat")) #with group labels
library("ggfortify")
autoplot(transformed.yy.pca,label=T,data=y2,colour="treatment")
# clustring and heatmap
heatmap(as.matrix(scale(t(transformed.yy))),scale="none")



#############################################
# for exon-based total-RNA
plotMDS(yy2)
#estimate teh dispersino
yy2 <- estimateGLMCommonDisp(yy2,design)
yy2 <- estimateGLMTrendedDisp(yy2,design)
yy2 <- estimateGLMTagwiseDisp(yy2,design)

#yy2$common.dispersion
plotBCV(yy2,cex=0.4)  ##there's some problem with the BCV plot; too few dots....

#corrlation matrix 
M <- cor(data.frame(cpm(yy2)),method="spearman")
corrplot.mixed(M,order="hclust")

# home-made PCA; log transform first on library-normalized datasets(many 0s; so try with log(x+1))
transformed.yy2 <- data.frame(cpm(yy2))
transformed.yy2 <- log(transformed.yy2+1)
transformed.yy2 <- as.data.frame(t(transformed.yy2))
transformed.yy2.pca <- prcomp(transformed.yy2,
                center=T,
                scale.=T)

#print(transformed.yy2.pca)
summary(transformed.yy2.pca)
plot(transformed.yy2.pca, type="l") # the first two pca cannot interpret the whole data (22% and 12%)

y2<- cbind(transformed.yy2,treatment=c("1487_basal" ,"2139_basal" ,"1487_treat", "2139_treat")) #with group labels
library("ggfortify")
autoplot(transformed.yy2.pca,label=T,data=y2,colour="treatment")
# clustring and heatmap
heatmap(as.matrix(scale(t(transformed.yy2))),scale="none",margin=c(5,5),cexCol=0.8)

```

##differential expression

Significance criteria: FDR < 0.1; log2_FC >0.6 or log2_FC<-0.6
*the outline dots in MA plot:* they are all (1487N=0, 2139N=0, 1487T=c1, 2139T=c2; 1487N=c1, 2139N=c2, 1487T=0, 2139T=0)
one example of middle grey dots around significant genes: (although log2cpm=7.4, log2fc=2)
1487_basal 2139_basal 1487_treat 2139_treat 
  27.96552   29.10831  610.55523   21.31747 (cpm)
examples of highly significant genes: (SELE, CXCL3, ICAM1, VCAM1, CXCL1)
     1487_basal 2139_basal 1487_treat 2139_treat (cpm)
316    43.72075   52.46266  1773.2767  1593.4062
6526   23.23896   30.46219   666.6454   976.1001
133    33.08597   92.06351  1366.3932  1736.0231
1645   73.26179  124.55651   822.5025  1367.3208
6155  590.03315  752.07762  3006.3408  4329.2488

examples of sig down genes:
     1487_basal 2139_basal 1487_treat 2139_treat (cpm)
16241   26.783881    5.753969    0.0000000    0.0000000
8732   127.223436   96.802070   23.4475275   48.3396254
1901    55.931046   51.108785   14.2524187    6.3051685
8601   185.517766  142.156886   87.8132893   60.0492241
1486    72.867912   51.785723   12.8731524   14.1115677
10405  177.640154  150.280137   70.3425825   75.6620223
16441   44.902389   38.923909    0.4597554    9.3076297
13143    8.271493   19.292720    0.0000000    0.0000000

example of sig up genes:
     1487_basal 2139_basal 1487_treat 2139_treat (cpm)
17079   64.2025391   63.2936612  145.742475  140.515184
1132     6.6959703    0.0000000   32.642636   36.930273
7160     0.0000000    0.0000000   11.034131   13.811322
17275  241.4488126  281.2675532  622.508867  348.885992
7015   381.2764283  290.0677414  502.052942  579.475012
16381    2.7571643   24.3697519   43.676767   64.252670


```{r echo=F}
fit <-  glmFit(yy,design)
lrt <- glmLRT(fit)

all_gene_list <- topTags(lrt,n=length(lrt))

gene_sig <- all_gene_list[all_gene_list$table$FDR<.1,]
gene_sig_up <- gene_sig[gene_sig$table$logFC>0,]
gene_sig_down <- gene_sig[gene_sig$table$logFC<0,]
gene_sigg_up <- gene_sig[gene_sig$table$logFC>0.6,]  ###275 siggup genes
gene_sigg_down <- gene_sig[gene_sig$table$logFC< -0.6,] ###103 siggdown genes


summary(gene_sig_up$table$logFC)
summary(gene_sig_down$table$logFC)

hist(all_gene_list$table$PValue,breaks=100,main="total rna intron",xlab="p value")
# the distribution of intron counts
ggplot(all_gene_list$table,aes(x=logCPM))+
  geom_histogram(aes(y=..density..),bins=50)+
  geom_density()+
  labs(title="total_rna_intron")
summary(all_gene_list$table$logCPM)

hist(gene_sig_up$table$logFC,breaks=100,main="Sig up genes",xlab="log_FC")
hist(gene_sig_down$table$logFC,breaks=100,main="Sig down genes",xlab="log_FC")
#plot MA plot
par(pch=19)
plot(lrt$table$logFC~lrt$table$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3),main="total rna intron",ylab="log2_FC",xlab="log2_CPM")
abline(h=0,col="magenta",lty=2)
points(gene_sig_up$table$logFC~gene_sig_up$table$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(gene_sig_down$table$logFC~gene_sig_down$table$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
abline(h=c(.6,-.6),col="magenta")
abline(a=4,b=1)
abline(a=-4,b=-1)
abline(v=c(7.4,7.5),h=c(1.9,2.1))
########################################################
# #analyze the variabillity of the ma plot
# idx.out <- ((all_gene_list$table$logCPM - all_gene_list$table$logFC + 4) < 0)
# rname<- rownames(all_gene_list$table[idx.out,])
# cpm(yy)[rname,]
# 
# #the bottom outliers
# idx.out <- ((all_gene_list$table$logCPM + all_gene_list$table$logFC + 4) < 0)
# rname<- rownames(all_gene_list$table[idx.out,])
# cpm(yy)[rname,]
# 
# # look at those grey dots surrounded by sig dots
# idx.out <- (all_gene_list$table$logCPM < 7.5 &all_gene_list$table$logCPM > 7.4 &all_gene_list$table$logFC < 2.1 &all_gene_list$table$logFC > 1.9 )
# rname<- rownames(all_gene_list$table[idx.out,])
# cpm(yy)[rname,]
# 
# # look at all the sig.down genes
# rname<- rownames(gene_sigg_down)
# cpm(yy)[rname,]
# 
# #look at all the sig.up genes
# rname<- rownames(gene_sigg_up)
# cpm(yy)[rname,]

###################################################
fit2 <-  glmFit(yy2,design)
lrt2 <- glmLRT(fit2)

all_gene_list2 <- topTags(lrt2,n=length(lrt2))

gene_sig2 <- all_gene_list2[all_gene_list2$table$FDR<.1,]
gene_sig_up2 <- gene_sig2[gene_sig2$table$logFC>0,]
gene_sig_down2 <- gene_sig2[gene_sig2$table$logFC<0,]
gene_sigg_up2 <- gene_sig2[gene_sig2$table$logFC>0.6,]  ###81 siggup genes
gene_sigg_down2 <- gene_sig2[gene_sig2$table$logFC< -0.6,] ### 15 siggdown genes


##compare with 370 sigg genes for  intron RNA-seq; only find 100 sigg genes for exon....

hist(all_gene_list2$table$PValue,breaks=100,main="total rna exon",xlab="p value")
# the distribution of intron counts
ggplot(all_gene_list2$table,aes(x=logCPM))+
  geom_histogram(aes(y=..density..),bins=50)+
  geom_density()+
  labs(title="total_rna_exon")
summary(all_gene_list2$table$logCPM)

hist(gene_sig_up2$table$logFC,breaks=100,main="Sig. up genes (exon)",xlab="log2_FC")
hist(gene_sig_down2$table$logFC,breaks=100,main="Sig. down genes (exon)",xlab="log2_FC")
#plot MA plot
par(pch=19)
plot(lrt2$table$logFC~lrt2$table$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3),main="total rna exon",ylab="log2_FC",xlab="log2_CPM")
abline(h=0,col="magenta",lty=2)
points(gene_sig_up2$table$logFC~gene_sig_up2$table$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(gene_sig_down2$table$logFC~gene_sig_down2$table$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
abline(h=c(.6,-.6),col="magenta")

#total intron and total exon reads signal histogram
ggplot(all_gene_list2$table,aes(x=logCPM))+
  geom_histogram(bins=50,alpha=0.4,fill="blue")+
  geom_histogram(data=all_gene_list$table,aes(x=logCPM),bins=50,alpha=0.4,fill="red")+
  labs(title="total_intron_vs_exon")

#total intron counts distribution for each sample
total_intron_samples<- log2(cpm(yy))
colnames(total_intron_samples) <- c('basal1487',"basal2139","treat1487","treat2139")
total_intron_samples <- as.data.frame(total_intron_samples)
total_exon_samples<- log2(cpm(yy2))
colnames(total_exon_samples) <- c('basal1487',"basal2139","treat1487","treat2139")
total_exon_samples <- as.data.frame(total_exon_samples)

ggplot(data=total_intron_samples,aes(x=basal1487))+
  geom_density(bins=50,alpha=0.3,color=color[2])+
  geom_density(aes(x=basal2139,color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=treat1487,color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=treat2139,color=color[2]),bins=50,alpha=0.3)+
  geom_density(data=total_exon_samples,aes(x=basal1487,color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=total_exon_samples,aes(x=basal2139,color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=total_exon_samples,aes(x=treat1487,color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=total_exon_samples,aes(x=treat2139,color=color[1]),bins=50,alpha=0.3)+
  labs(title="total_RNA samples",x="log2cpm")+
  scale_color_discrete(labels=c("intron","exon"),name="Type")

  

```


```{r echo=F}
barplot(c(nrow(gene_sig),nrow(gene_sigg_up),nrow(gene_sigg_down)),main="intron vs exon RNA analysis", ylab = "numbers of significant gene",ylim=c(0,400))

barplot(c(nrow(gene_sig),nrow(gene_sigg_up),nrow(gene_sigg_down),nrow(gene_sig2),nrow(gene_sigg_up2),nrow(gene_sigg_down2)),main="intron vs exon RNA analysis", ylab = "numbers of significant gene",ylim=c(0,400))
```

```{r}
geneN <- seq(529)
analysis <- factor(c(rep("intron",389),rep("exon",140)))
change <- factor(c(rep("significantly upregulated",275),rep("significantly downregulated",103),rep("below foldchange threshold",11),rep("significantly upregulated",81),rep("significantly downregulated",15),rep("below foldchange threshold",44)))
d <- data.frame(geneN=geneN,analysis=analysis,change=change)
library(ggplot2)
ggplot(d, aes(x=analysis,fill=change))+
  geom_bar(width=.5)+
  ggtitle("exon vs intron RNA analysis")

  

```


**Significant genes**
Sig_intron_genes|I_increased|I_decreased|Sig_exon_genes|E_increased|E_decreased 
---|---|---|---|---|---|
`r nrow(gene_sig)`|`r nrow(gene_sigg_up)`|`r nrow(gene_sigg_down)`|`r nrow(gene_sig2)`|`r nrow(gene_sigg_up2)`|`r nrow(gene_sigg_down2)`



##volcano plot

```{r echo=F}
#intron_analysis, significant and logfc>0.6 genes are denoted
par(pch=19)
log_neg_pvalue <-  -log10(all_gene_list$table$PValue)
plot(log_neg_pvalue~all_gene_list$table$logFC,cex=0.6,col=rgb(0,0,0,alpha=0.3),xlab="log2_FC",ylab="-log2(pval)",main="total rna intron")
log_neg_sigup_pval <- -log10(gene_sigg_up$table$PValue)
points(log_neg_sigup_pval~gene_sigg_up$table$logFC,col=rgb(0.7,0,0,alpha=0.3),cex=0.6)
log_neg_sigdown_pval <- -log10(gene_sigg_down$table$PValue)
points(log_neg_sigdown_pval~gene_sigg_down$table$logFC,col=rgb(0,0,0.7,alpha=0.3),cex=0.6)
#text(gene_sigg_up$table$logFC,log_neg_sigup_pval, labels=gene_sigg_up$table$genes, cex= 0.5,font=2,pos=4)
#text(gene_sigg_down$table$logFC,log_neg_sigdown_pval, labels=gene_sigg_down$table$genes, cex= 0.5,font=2,pos=3)
```



**How intron-sig genes and exon_sig genes ovelap**

```{r echo=F}
library(grid)
library(futile.logger)
library(VennDiagram)
intron_siggup_gene_list <- as.character(gene_sigg_up$table$genes)
intron_siggdown_gene_list <- as.character(gene_sigg_down$table$genes)
exon_siggup_gene_list <- as.character(gene_sigg_up2$table$genes)
exon_siggdown_gene_list <- as.character(gene_sigg_down2$table$genes)
# venn.diagram(
#   x = list(exon_siggup_gene_list, polya_exon_siggup_gene_list),
#   category.names = c("exon_sig_up", "polya_exon_sig_up"),
#   filename = "total_vs_polya_exon_up"
#   
#   )
#write.table(intron_siggup_gene_list,file="intron_sig_up_gene_list",sep="\t",row.names = F,col.names = F)
#write.table(intron_siggdown_gene_list,file="intron_sig_down_gene_list",sep="\t",row.names=F, col.names=F)

# the overlap between intron and exon
up_overlap <- sum(exon_siggup_gene_list %in% intron_siggup_gene_list)
print("sig_up gene overlap")
length(intron_siggup_gene_list)-up_overlap;up_overlap; length(exon_siggup_gene_list)-up_overlap

down_overlap <- sum(exon_siggdown_gene_list %in% intron_siggdown_gene_list)
print("sig_down gene overlap")
length(intron_siggdown_gene_list)-down_overlap;down_overlap; length(exon_siggdown_gene_list)-down_overlap




```

# Poly RNA analysis

##countable analysis
the countable is generated by iRNA_seq pipeline, so I will get the edgeR anaylsis manually
and this is on polya-seg sample

```{r echo=F}
library(edgeR)
counttable_path <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/polya_basal_treatment_intron_counttable.txt"
counttable_path2 <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/polya_basal_treatment_exon_counttable.txt"
intron_table <- read.table(counttable_path,sep="\t",header=T)
exon_table <- read.table(counttable_path2,sep="\t",header=T)
colnames(intron_table)[9:12] <- c('1487_basal',"2139_basal","1487_treat","2139_treat")
polya_yy <- DGEList(counts=intron_table[9:12],genes=intron_table$Symbol,group=factor(c("N","N","T","T")))

colnames(exon_table)[9:12] <- c('1487_basal',"2139_basal","1487_treat","2139_treat")
polya_yy2 <- DGEList(counts=exon_table[9:12],genes=exon_table$Symbol,group=factor(c("N","N","T","T")))
```
(seems intron reads make up 10% of the whole reads)

**lib.size for exon only 20 million each????? much fewer than expected**
begin analysis
1 million for intron reads; 10 million for exon reads
```{r echo=F}
o <- order(rowSums(polya_yy$counts),decreasing = T)
polya_yy <- polya_yy[o,]
idx <- rowSums(cpm(polya_yy)>2)>=2  # ~10000 genes left
polya_yy <- polya_yy[idx,]
polya_yy$samples$lib.size <- colSums(polya_yy$counts)
polya_yy <- calcNormFactors(polya_yy)
polya_yy$samples

o <- order(rowSums(polya_yy2$counts),decreasing = T)
polya_yy2 <- polya_yy2[o,]
idx <- rowSums(cpm(polya_yy2)>2)>=2  # ~10000 genes left
polya_yy2 <- polya_yy2[idx,]
polya_yy2$samples$lib.size <- colSums(polya_yy2$counts)
polya_yy2 <- calcNormFactors(polya_yy2)
polya_yy2$samples
```
```{r echo=F}
plotMDS(polya_yy)
patient <- factor(c("1487","2139","1487","2139"))
tissue <- factor(c("N","N","T","T"))
design <- model.matrix(~patient+tissue)
rownames(design) <- c("1487N","2139N","1487T","2139T")

#estimate teh dispersino
polya_yy <- estimateGLMCommonDisp(polya_yy,design)
polya_yy <- estimateGLMTrendedDisp(polya_yy,design)#~10000 genes left after filtering
polya_yy <- estimateGLMTagwiseDisp(polya_yy,design)

#polya_yy$common.dispersion
plotBCV(polya_yy,cex=0.4)  ##there's some problem with the BCV plot; too few dots....
#corrlation matrix 
M <- cor(data.frame(cpm(polya_yy)),method="spearman")
corrplot.mixed(M,order="hclust")

# home-made PCA; log transform first on library-normalized datasets(many 0s; so try with log(x+1))
transformed.polya_yy <- data.frame(cpm(polya_yy))
transformed.polya_yy <- log(transformed.polya_yy+1)
transformed.polya_yy <- as.data.frame(t(transformed.polya_yy))
transformed.polya_yy.pca <- prcomp(transformed.polya_yy,
                center=T,
                scale.=T)

# print(transformed.polya_yy.pca)
summary(transformed.polya_yy.pca)
plot(transformed.polya_yy.pca, type="l") # the first two pca cannot interpret the whole data (22% and 12%)

y2<- cbind(transformed.polya_yy,treatment=c("1487_basal" ,"2139_basal" ,"1487_treat", "2139_treat")) #with group labels
library("ggfortify")
autoplot(transformed.polya_yy.pca,label=T,data=y2,colour="treatment")
# clustring and heatmap
heatmap(as.matrix(scale(t(transformed.polya_yy))),scale="none")

#########################################
#ployA_ exon analysis
plotMDS(polya_yy2)


#estimate teh dispersino
polya_yy2 <- estimateGLMCommonDisp(polya_yy2,design)
polya_yy2 <- estimateGLMTrendedDisp(polya_yy2,design)
polya_yy2 <- estimateGLMTagwiseDisp(polya_yy2,design)

polya_yy2$common.dispersion
plotBCV(polya_yy2,cex=0.4)  ##there's some problem with the BCV plot; too few dots....


#corrlation matrix 
M <- cor(data.frame(cpm(polya_yy2)),method="spearman")
corrplot.mixed(M,order="hclust")

# home-made PCA; log transform first on library-normalized datasets(many 0s; so try with log(x+1))
transformed.polya_yy2 <- data.frame(cpm(polya_yy2))
transformed.polya_yy2 <- log(transformed.polya_yy2+1)
transformed.polya_yy2 <- as.data.frame(t(transformed.polya_yy2))
transformed.polya_yy2.pca <- prcomp(transformed.polya_yy2,
                center=T,
                scale.=T)

# print(transformed.polya_yy2.pca)
summary(transformed.polya_yy2.pca)
plot(transformed.polya_yy2.pca, type="l") # the first two pca cannot interpret the whole data (22% and 12%)

y2<- cbind(transformed.polya_yy2,treatment=c("1487_basal" ,"2139_basal" ,"1487_treat", "2139_treat")) #with group labels
library("ggfortify")
autoplot(transformed.polya_yy2.pca,label=T,data=y2,colour="treatment")
# clustring and heatmap
heatmap(as.matrix(scale(t(transformed.polya_yy2))),scale="none")


```

##differential expression

```{r echo=F}
fit <-  glmFit(polya_yy,design)
lrt <- glmLRT(fit)

polya_all_gene_list <- topTags(lrt,n=length(lrt))
logfc_threshold <- 0.6
polya_gene_sig <- polya_all_gene_list[polya_all_gene_list$table$FDR<.1,]
polya_gene_sig_up <- polya_gene_sig[polya_gene_sig$table$logFC>0,]
polya_gene_sig_down <- polya_gene_sig[polya_gene_sig$table$logFC<0,]
polya_gene_sigg_up <- polya_gene_sig[polya_gene_sig$table$logFC>logfc_threshold,]  ###332 siggup genes
polya_gene_sigg_down <- polya_gene_sig[polya_gene_sig$table$logFC< -logfc_threshold,] ###142 siggdown genes



hist(polya_all_gene_list$table$PValue,breaks=100)
hist(polya_gene_sig$table$logFC,breaks=100)

hist(polya_gene_sig_up$table$logFC,breaks=100)
hist(polya_gene_sig_down$table$logFC,breaks=100)
#plot MA plot
par(pch=19)
plot(lrt$table$logFC~lrt$table$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3))
abline(h=0,col="magenta",lty=2)
points(polya_gene_sig_up$table$logFC~polya_gene_sig_up$table$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(polya_gene_sig_down$table$logFC~polya_gene_sig_down$table$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
abline(h=c(.6,-.6),col="magenta")


#################################
#polyA_exon
fit2 <-  glmFit(polya_yy2,design)
lrt2 <- glmLRT(fit2)

polya_all_gene_list2 <- topTags(lrt2,n=length(lrt2))

polya_gene_sig2 <- polya_all_gene_list2[polya_all_gene_list2$table$FDR<.1,]
polya_gene_sig_up2 <- polya_gene_sig2[polya_gene_sig2$table$logFC>0,]
polya_gene_sig_down2 <- polya_gene_sig2[polya_gene_sig2$table$logFC<0,]
polya_gene_sigg_up2 <- polya_gene_sig2[polya_gene_sig2$table$logFC>logfc_threshold,]  ###124 siggup genes
polya_gene_sigg_down2 <- polya_gene_sig2[polya_gene_sig2$table$logFC< -logfc_threshold,] ### 21 siggdown genes


summary(polya_gene_sig_up2$table$logFC)
summary(polya_gene_sig_down2$table$logFC)  ##compare with 370 sigg genes for  intron RNA-seq; only find 100 sigg genes for exon....

hist(polya_all_gene_list2$table$PValue,breaks=100)
hist(polya_gene_sig_up2$table$logFC,breaks=100)

#plot MA plot
par(pch=19)
plot(lrt2$table$logFC~lrt2$table$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3))
abline(h=0,col="magenta",lty=2)
points(polya_gene_sig_up2$table$logFC~polya_gene_sig_up2$table$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(polya_gene_sig_down2$table$logFC~polya_gene_sig_down2$table$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
abline(h=c(.6,-.6),col="magenta")


#ploya intron and polya exon reads signal histogram
ggplot(polya_all_gene_list2$table,aes(x=logCPM))+
  geom_histogram(bins=50,alpha=0.4,fill="blue")+
  geom_histogram(data=polya_all_gene_list$table,aes(x=logCPM),bins=50,alpha=0.4,fill="red")+
  labs(title="total_intron_vs_exon")

#polya intron counts distribution for each sample
polya_intron_samples<- log2(cpm(polya_yy))
colnames(polya_intron_samples) <- c('basal1487',"basal2139","treat1487","treat2139")
polya_intron_samples <- as.data.frame(polya_intron_samples)
polya_exon_samples<- log2(cpm(yy2))
colnames(polya_exon_samples) <- c('basal1487',"basal2139","treat1487","treat2139")
polya_exon_samples <- as.data.frame(polya_exon_samples)
ggplot(data=polya_intron_samples,aes(x=basal1487))+
  geom_histogram(bins=50,alpha=0.3,fill=color[2])+
  geom_histogram(aes(x=basal2139),bins=50,alpha=0.3,fill=color[2])+
  geom_histogram(aes(x=treat1487),bins=50,alpha=0.3,fill=color[2])+
  geom_histogram(aes(x=treat2139),bins=50,alpha=0.3,fill=color[2])+
  geom_histogram(data=polya_exon_samples,aes(x=basal1487),bins=50,alpha=0.3,fill=color[1])+
  geom_histogram(data=polya_exon_samples,aes(x=basal2139),bins=50,alpha=0.3,fill=color[1])+
  geom_histogram(data=polya_exon_samples,aes(x=treat1487),bins=50,alpha=0.3,fill=color[1])+
  geom_histogram(data=polya_exon_samples,aes(x=treat2139),bins=50,alpha=0.3,fill=color[1])

ggplot(data=polya_intron_samples,aes(x=basal1487))+
  geom_density(bins=50,alpha=0.3,color=color[2])+
  geom_density(aes(x=basal2139,color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=treat1487,color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=treat2139,color=color[2]),bins=50,alpha=0.3)+
  geom_density(data=polya_exon_samples,aes(x=basal1487,color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=polya_exon_samples,aes(x=basal2139,color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=polya_exon_samples,aes(x=treat1487,color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=polya_exon_samples,aes(x=treat2139,color=color[1]),bins=50,alpha=0.3)+
  labs(title="polya_RNA samples",x="log2cpm")+
  scale_color_discrete(labels=c("intron","exon"),name="Type")

```
how intron-sig genes and exon_sig genes ovelap

```{r echo=F}
library(grid)
library(futile.logger)
library(VennDiagram)
polya_intron_siggup_gene_list <- as.character(polya_gene_sigg_up$table$genes)
polya_intron_siggdown_gene_list <- as.character(polya_gene_sigg_down$table$genes)
polya_exon_siggup_gene_list <- as.character(polya_gene_sigg_up2$table$genes)
polya_exon_siggdown_gene_list <- as.character(polya_gene_sigg_down2$table$genes)

#write.table(intron_siggup_gene_list,file="intron_sig_up_gene_list",sep="\t",row.names = F,col.names = F)
#write.table(intron_siggdown_gene_list,file="intron_sig_down_gene_list",sep="\t",row.names=F, col.names=F)

# the overlap between intron and exon
polya_up_overlap <- sum(polya_exon_siggup_gene_list %in% polya_intron_siggup_gene_list)
print("polya_sig_up gene overlap");
length(polya_intron_siggup_gene_list)-polya_up_overlap;polya_up_overlap; length(polya_exon_siggup_gene_list)-polya_up_overlap

polya_down_overlap <- sum(polya_exon_siggdown_gene_list %in% polya_intron_siggdown_gene_list)
print("polya_sig_down gene overlap");
length(polya_intron_siggdown_gene_list)-polya_down_overlap;down_overlap; length(polya_exon_siggdown_gene_list)-polya_down_overlap

```



for the `polyA analysis`, it's similar. increased genes for both intron/ exon can be even larger overlapped
but still rare overlap for the down genes


look at the correlation between total rna and polya library; after filtering out, the effective exon/ intron reads are similar, also the filtered gene list number are similar

For the increased genes: 

```{r echo=F}
#overlap between paired groups
print("input intron_analysis genes for comparison")
sum(all_gene_list$table$genes %in% polya_all_gene_list$table$genes)
print("input exon_analysis genes for comparison")
sum(all_gene_list2$table$genes %in% polya_all_gene_list2$table$genes)

intron_gene_name <- as.character(all_gene_list[all_gene_list$table$genes %in% polya_all_gene_list$table$genes]$table$genes)
exon_gene_name <- as.character(all_gene_list2[all_gene_list2$table$genes %in% polya_all_gene_list2$table$genes]$table$genes)

total_intron_FC <- vector("numeric",length(intron_gene_name))
polya_intron_FC <- vector("numeric",length(intron_gene_name))

total_exon_FC <- vector("numeric",length(exon_gene_name))
polya_exon_FC <- vector("numeric",length(exon_gene_name))

for ( i in seq(length(intron_gene_name))) {
  total_intron_FC[i]=all_gene_list[all_gene_list$table$genes==intron_gene_name[i],]$table$logFC
  polya_intron_FC[i]=polya_all_gene_list[polya_all_gene_list$table$genes==intron_gene_name[i],]$table$logFC
}

for ( i in seq(length(exon_gene_name))) {
  total_exon_FC[i]=all_gene_list2[all_gene_list2$table$genes==exon_gene_name[i],]$table$logFC
  polya_exon_FC[i]=polya_all_gene_list2[polya_all_gene_list2$table$genes==exon_gene_name[i],]$table$logFC
}

#intron_sigg_gene_list <- c(intron_siggup_gene_list,intron_siggdown_gene_list)
#358 out of 383 intron_sigg genes can be found in input of polya_all_gene_list (intron) ; just plot them separately
intron_siggup_overlap_gene <- intron_siggup_gene_list[intron_siggup_gene_list %in% polya_all_gene_list$table$genes]
intron_siggdown_overlap_gene <- intron_siggdown_gene_list[intron_siggdown_gene_list %in% polya_all_gene_list$table$genes]


total_intron_siggup_FC <- vector("numeric",length(intron_siggup_overlap_gene))
polya_intron_siggup_FC <- vector("numeric",length(intron_siggup_overlap_gene))

total_intron_siggdown_FC <- vector("numeric",length(intron_siggdown_overlap_gene))
polya_intron_siggdown_FC <- vector("numeric",length(intron_siggdown_overlap_gene))

for ( i in seq(length(intron_siggup_overlap_gene))) {
  total_intron_siggup_FC[i]=all_gene_list[all_gene_list$table$genes==intron_siggup_overlap_gene[i],]$table$logFC
  polya_intron_siggup_FC[i]=polya_all_gene_list[polya_all_gene_list$table$genes==intron_siggup_overlap_gene[i],]$table$logFC
}

for ( i in seq(length(intron_siggdown_overlap_gene))) {
  total_intron_siggdown_FC[i]=all_gene_list[all_gene_list$table$genes==intron_siggdown_overlap_gene[i],]$table$logFC
  polya_intron_siggdown_FC[i]=polya_all_gene_list[polya_all_gene_list$table$genes==intron_siggdown_overlap_gene[i],]$table$logFC
}


##for exon genes
exon_siggup_overlap_gene <- exon_siggup_gene_list[exon_siggup_gene_list %in% polya_all_gene_list$table$genes]
exon_siggdown_overlap_gene <- exon_siggdown_gene_list[exon_siggdown_gene_list %in% polya_all_gene_list$table$genes]


total_exon_siggup_FC <- vector("numeric",length(exon_siggup_overlap_gene))
polya_exon_siggup_FC <- vector("numeric",length(exon_siggup_overlap_gene))

total_exon_siggdown_FC <- vector("numeric",length(exon_siggdown_overlap_gene))
polya_exon_siggdown_FC <- vector("numeric",length(exon_siggdown_overlap_gene))

for ( i in seq(length(exon_siggup_overlap_gene))) {
  total_exon_siggup_FC[i]=all_gene_list2[all_gene_list2$table$genes==exon_siggup_overlap_gene[i],]$table$logFC
  polya_exon_siggup_FC[i]=polya_all_gene_list2[polya_all_gene_list2$table$genes==exon_siggup_overlap_gene[i],]$table$logFC
}

for ( i in seq(length(exon_siggdown_overlap_gene))) {
  total_exon_siggdown_FC[i]=all_gene_list2[all_gene_list2$table$genes==exon_siggdown_overlap_gene[i],]$table$logFC
  polya_exon_siggdown_FC[i]=polya_all_gene_list2[polya_all_gene_list2$table$genes==exon_siggdown_overlap_gene[i],]$table$logFC
}





par(pch=19)
plot(total_intron_FC~polya_intron_FC,cex=0.6,col=rgb(0,0,0,alpha=0.3),main="highlighted total rna intron sig. genes")
intron_lm <- lm(total_intron_FC~polya_intron_FC)
abline(-0.10115,0.87802,col="magenta")
text(-2,10,"R^2=0.12")
# plot on the basis of intron_sigg_up or intron_sigg_down genes total
points(total_intron_siggup_FC~polya_intron_siggup_FC,col=rgb(1,0,0,alpha=0.3),cex=0.6)
points(total_intron_siggdown_FC~polya_intron_siggdown_FC,col=rgb(0,0,1,alpha=0.3),cex=0.6)
abline(h=c(-0.6,0.6),col="magenta",lty=2)
abline(v=c(-0.6,0.6),col="magenta",lty=2)

###plot based on polya indentified intron or exon genes
#448 out of 478 can be found in the filtered input of total rnaseq library
polya_intron_sigg_gene <- c(polya_intron_siggup_gene_list,polya_intron_siggdown_gene_list)
polya_intron_sigg_overlap <- polya_intron_sigg_gene[polya_intron_sigg_gene %in% all_gene_list$table$genes]


total_intron_sigg_FC <- vector("numeric",length(polya_intron_sigg_overlap))
polya_intron_sigg_FC <- vector("numeric",length(polya_intron_sigg_overlap))

for ( i in seq(length(polya_intron_sigg_overlap))) {
  total_intron_sigg_FC[i]=all_gene_list[all_gene_list$table$genes==polya_intron_sigg_overlap[i],]$table$logFC
  polya_intron_sigg_FC[i]=polya_all_gene_list[polya_all_gene_list$table$genes==polya_intron_sigg_overlap[i],]$table$logFC
}

par(pch=19)
plot(total_intron_FC~polya_intron_FC,cex=0.6,col=rgb(0,0,0,alpha=0.3),main="highlighted polyA rna intron sig. genes")
intron_lm <- lm(total_intron_FC~polya_intron_FC)
abline(-0.10115,0.87802,col="magenta")
text(-2,10,"R^2=0.12")
# plot on the basis of intron_sigg_up or intron_sigg_down genes total
points(total_intron_sigg_FC~polya_intron_sigg_FC,col=rgb(1,0,0,alpha=0.3),cex=0.6)
abline(h=c(-0.6,0.6),col="magenta",lty=2)
abline(v=c(-0.6,0.6),col="magenta",lty=2)




par(pch=19)
plot(total_exon_FC~polya_exon_FC,cex=0.6,col=rgb(0,0,0,alpha=0.3),main="highlighted total rna exon sig. genes")
exon_lm <- lm(total_exon_FC~polya_exon_FC)
abline(-0.017919,0.884613,col="magenta")
text(-1.5,6,"R^2=0.49")
# plot on the basis of exon_sigg_up or exon_sigg_down genes  total
points(total_exon_siggup_FC~polya_exon_siggup_FC,col=rgb(1,0,0,alpha=0.3),cex=0.6)
points(total_exon_siggdown_FC~polya_exon_siggdown_FC,col=rgb(0,0,1,alpha=0.3),cex=0.6)
abline(h=c(-0.6,0.6),col="magenta",lty=2)
abline(v=c(-0.6,0.6),col="magenta",lty=2)













```

##distribution of gene counts


```{r include=F}
hist(all_gene_list$table$logCPM,breaks=100)
d <- density(all_gene_list$table$logCPM)
line(d)
hist(polya_all_gene_list$table$logCPM,breaks=100)

```



