---
title: "edgR-fC-polya_analysis"
output: html_document
---
#countable analysis
the countable is generated by iRNA_seq pipeline, so I will get the edgeR anaylsis manually
and this is on polya-seg sample

```{r echo=F}
library(edgeR)
counttable_path <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/polya_basal_treatment_intron_counttable.txt"
counttable_path2 <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/polya_basal_treatment_exon_counttable.txt"
intron_table <- read.table(counttable_path,sep="\t",header=T)
exon_table <- read.table(counttable_path2,sep="\t",header=T)
colnames(intron_table)[9:12] <- c('1487_basal',"2139_basal","1487_treat","2139_treat")
polya_yy <- DGEList(counts=intron_table[9:12],genes=intron_table$Symbol,group=factor(c("N","N","T","T")))

colnames(exon_table)[9:12] <- c('1487_basal',"2139_basal","1487_treat","2139_treat")
polya_yy2 <- DGEList(counts=exon_table[9:12],genes=exon_table$Symbol,group=factor(c("N","N","T","T")))
```
(seems intron reads make up 10% of the whole reads)

**lib.size for exon only 20 million each????? much fewer than expected**
begin analysis
1 million for intron reads; 10 million for exon reads
```{r echo=F}
o <- order(rowSums(polya_yy$counts),decreasing = T)
polya_yy <- polya_yy[o,]
idx <- rowSums(cpm(polya_yy)>4)>=2  # ~10000 genes being filtered out
polya_yy <- polya_yy[idx,]
polya_yy$samples$lib.size <- colSums(polya_yy$counts)
polya_yy <- calcNormFactors(polya_yy)
polya_yy$samples

o <- order(rowSums(polya_yy$counts),decreasing = T)
polya_yy2 <- polya_yy2[o,]
idx <- rowSums(cpm(polya_yy2)>4)>=2  # ~10000 genes being filtered out
polya_yy2 <- polya_yy2[idx,]
polya_yy2$samples$lib.size <- colSums(polya_yy2$counts)
polya_yy2 <- calcNormFactors(polya_yy2)
polya_yy2$samples
```
data exploratrion

```{r echo=F}
plotMDS(polya_yy)
patient <- factor(c("1487","2139","1487","2139"))
tissue <- factor(c("N","N","T","T"))
design <- model.matrix(~patient+tissue)
rownames(design) <- c("1487N","2139N","1487T","2139T")
design

#estimate teh dispersion
polya_yy <- estimateGLMCommonDisp(polya_yy,design)
polya_yy <- estimateGLMTrendedDisp(polya_yy,design)
polya_yy <- estimateGLMTagwiseDisp(polya_yy,design)

polya_yy$common.dispersion
plotBCV(polya_yy,cex=0.4)  ##there's some problem with the BCV plot; too few dots....




plotMDS(polya_yy2)


#estimate teh dispersino
polya_yy2 <- estimateGLMCommonDisp(polya_yy2,design)
polya_yy2 <- estimateGLMTrendedDisp(polya_yy2,design)
polya_yy2 <- estimateGLMTagwiseDisp(polya_yy2,design)

polya_yy2$common.dispersion
plotBCV(polya_yy2,cex=0.4)  ##there's some problem with the BCV plot; too few dots....

```

differential expression
```{r echo=F}
fit <-  glmFit(polya_yy,design)
lrt <- glmLRT(fit)

polya_all_gene_list <- topTags(lrt,n=length(lrt))
logfc_threshold <- 0.6
polya_gene_sig <- polya_all_gene_list[polya_all_gene_list$table$FDR<.1,]
polya_gene_sig_up <- polya_gene_sig[polya_gene_sig$table$logFC>0,]
polya_gene_sig_down <- polya_gene_sig[polya_gene_sig$table$logFC<0,]
polya_gene_sigg_up <- polya_gene_sig[polya_gene_sig$table$logFC>logfc_threshold,]  ###332 siggup genes
polya_gene_sigg_down <- polya_gene_sig[polya_gene_sig$table$logFC< -logfc_threshold,] ###142 siggdown genes


summary(polya_gene_sig_up$table$logFC)
summary(polya_gene_sig_down$table$logFC)

hist(polya_all_gene_list$table$PValue,breaks=100)
hist(polya_gene_sig$table$logFC,breaks=100)

hist(polya_gene_sig_up$table$logFC,breaks=100)
hist(polya_gene_sig_down$table$logFC,breaks=100)
#plot MA plot
par(pch=19)
plot(lrt$table$logFC~lrt$table$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3))
abline(h=0,col="magenta",lty=2)
points(polya_gene_sig_up$table$logFC~polya_gene_sig_up$table$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(polya_gene_sig_down$table$logFC~polya_gene_sig_down$table$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
abline(h=c(.6,-.6),col="magenta")




fit2 <-  glmFit(polya_yy2,design)
lrt2 <- glmLRT(fit2)

polya_all_gene_list2 <- topTags(lrt2,n=length(lrt2))

polya_gene_sig2 <- polya_all_gene_list2[polya_all_gene_list2$table$FDR<.1,]
polya_gene_sig_up2 <- polya_gene_sig2[polya_gene_sig2$table$logFC>0,]
polya_gene_sig_down2 <- polya_gene_sig2[polya_gene_sig2$table$logFC<0,]
polya_gene_sigg_up2 <- polya_gene_sig2[polya_gene_sig2$table$logFC>logfc_threshold,]  ###124 siggup genes
polya_gene_sigg_down2 <- polya_gene_sig2[polya_gene_sig2$table$logFC< -logfc_threshold,] ### 21 siggdown genes


summary(polya_gene_sig_up2$table$logFC)
summary(polya_gene_sig_down2$table$logFC)  ##compare with 370 sigg genes for  intron RNA-seq; only find 100 sigg genes for exon....

hist(polya_all_gene_list2$table$PValue,breaks=100)
hist(polya_gene_sig2$table$logFC,breaks=100)

hist(polya_gene_sig_up2$table$logFC,breaks=100)
hist(polya_gene_sig_down2$table$logFC,breaks=100)
#plot MA plot
par(pch=19)
plot(lrt2$table$logFC~lrt2$table$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3))
abline(h=0,col="magenta",lty=2)
points(polya_gene_sig_up2$table$logFC~polya_gene_sig_up2$table$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(polya_gene_sig_down2$table$logFC~polya_gene_sig_down2$table$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
abline(h=c(.6,-.6),col="magenta")
```
how intron-sig genes and exon_sig genes ovelap

```{r echo=F}
library(grid)
library(futile.logger)
library(VennDiagram)
polya_intron_siggup_gene_list <- as.character(polya_gene_sigg_up$table$genes)
polya_intron_siggdown_gene_list <- as.character(polya_gene_sigg_down$table$genes)
polya_exon_siggup_gene_list <- as.character(polya_gene_sigg_up2$table$genes)
polya_exon_siggdown_gene_list <- as.character(polya_gene_sigg_down2$table$genes)

#write.table(polya_intron_siggup_gene_list,file="polya_intron_siggup_gene_list",sep="\t",row.names = F,col.names = F)
#write.table(polya_intron_siggdown_gene_list,file="polya_intron_siggdown_gene_list",sep="\t",row.names=F, col.names=F)

# the overlap between intron and exon
polya_up_overlap <- sum(polya_exon_siggup_gene_list %in% polya_intron_siggup_gene_list)
print("polya_sig_up gene overlap");
length(polya_intron_siggup_gene_list)-polya_up_overlap;polya_up_overlap; length(polya_exon_siggup_gene_list)-polya_up_overlap

polya_down_overlap <- sum(polya_exon_siggdown_gene_list %in% polya_intron_siggdown_gene_list)
print("polya_sig_down gene overlap");
length(polya_intron_siggdown_gene_list)-polya_down_overlap;down_overlap; length(polya_exon_siggdown_gene_list)-polya_down_overlap

```