---
title: "iRNA-seq_improved"
author: "DW"
date: "April 6, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Improved version of iRNA-seq analysis
this code is to cut out the redundancy of original code by packaging functions

*Function part*

```{r}
library(edgeR)
## read a countable and create a DGElist out from it
DGE_generater <- function(countable_path){
  #input: path of a gene counttable generated by iRNA-seq; output: one DGEList object
  # counttable_path <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_intron_counttable.txt"
  # counttable_path2 <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_exon_counttable.txt"
  intron_table <- read.table(countable_path,sep="\t",header=T)
  colnames(intron_table)[9:12] <- c('basal1487',"basal2139","treat1487","treat2139")
  yy <- DGEList(counts=intron_table[9:12],genes=intron_table$Symbol,group=factor(c("N","N","T","T")))
  return(yy)
}

## preprocess the DGElist and print some qc figures
normalize <- function(yy,design){
  #input:DGElist object and a design object; output: a normalized DGElist object
  #reordering and filtering the yy; calculate the lib.size and the dispersions;plot the default "PCA" and the BCV
  o <- order(rowSums(yy$counts),decreasing = T)
  yy <- yy[o,]
  idx <- rowSums(cpm(yy)>2)>=2  # ~10000 genes left
  yy <- yy[idx,]
  yy$samples$lib.size <- colSums(yy$counts)
  yy <- calcNormFactors(yy)
  # yy$samples  # to show the lib.size and the norm.factoers
  plotMDS(yy,cex=0.7)
  yy <- estimateGLMCommonDisp(yy,design)
  yy <- estimateGLMTrendedDisp(yy,design)
  yy <- estimateGLMTagwiseDisp(yy,design)
  plotBCV(yy,cex=0.4)
  return(yy)
}

## a bunch of qc_plot to know about the intrinsic data structure (correlation matrix, PCA, correlation heatmap)
qc_bundle <- function(yy, correlation.method="spearman",transform.method="log+1"){
  #plot correlation matrix, PCA, correlation heatmap between samples
  #input: a normalized DGElist generated by normalize function, a character to say the correlation method used in correlatin matirx, a character to show how to transform the data
  #this function doesn't change yy
  library(corrplot)
  library(ggfortify)
  library(gplots)
  transformed.yy <- data.frame(cpm(yy))
  if (transform.method=="log+1"){
    transformed.yy <- log(transformed.yy+1) 
  }else if (transform.method=="log"){
    transformed.yy <- log(transformed.yy) 
  }else {
    return(1)
  }
  #correlatin matirx
  M <- cor(transformed.yy,method=correlation.method)
  corrplot.mixed(M,order="hclust")
  #PCA
  transformed.yy.pca <- prcomp(t(transformed.yy),
                center=T,
                scale.=T)
  #print(transformed.yy.pca)
  #summary(transformed.yy.pca)
  y2<- cbind(t(transformed.yy),treatment=c("basal1487" ,"basal2139" ,"treat1487", "treat2139")) #with group labels
  plot(transformed.yy.pca, type="l")
  #heatmap
  sample_dist <- dist(1-cor(transformed.yy)) #perform euclidean distance on the normalized and transformed data, pearson correlation; 1-pearson correlation = distance
  hc <- hclust(sample_dist,method="average")
  heatmap.2(cor(transformed.yy),Rowv=as.dendrogram(hc),symm=T,trace="none",margins=c(7,7),cexRow = 0.9,cexCol = 0.9)
  #plot PCA
  fig<- autoplot(transformed.yy.pca,label=T,data=y2,colour="treatment")
  print(fig)
}

## GLM fitting and LRT test
GLM_fitting <- function(yy,design,analysis.name, fdr=0.1,logfc=0.6){
  #input: a normalized DGElist generated by normalize function, a design matrix, the title used in the plot(e.g. "total intron"), the FDR & logFC threshold used in the analysis
  #output: a list containing all_gene_list (topTag object) and gene_sigg_up & gene_sigg_down
  #this function is to fitting yy using thresholds to get sig up and sig-down genes and plot p_value distribution, logCPM distribution, MA plot and volcano plot
  library(ggplot2)
  library(RColorBrewer)
  color <- brewer.pal(n=3,name="Set1")
  
  fit <-  glmFit(yy,design)
  lrt <- glmLRT(fit)
  
  all_gene_list <- topTags(lrt,n=length(lrt))
  
  gene_sig <- all_gene_list[all_gene_list$table$FDR<fdr,]
  gene_sig_up <- gene_sig[gene_sig$table$logFC>0,]
  gene_sig_down <- gene_sig[gene_sig$table$logFC<0,]
  gene_sigg_up <- gene_sig[gene_sig$table$logFC>logfc,]  ##efksdfkl;#275 siggup genes
  gene_sigg_down <- gene_sig[gene_sig$table$logFC< -logfc,] ###103 siggdown genes
  #the distribution of pvalue
  fig1<- ggplot(all_gene_list$table,aes(x=PValue))+
    geom_histogram(bins=100)+
    geom_vline(xintercept = 0.1, linetype="dashed")
    labs(title=analysis.name)
  # the distribution of counts
  fig2 <- ggplot(all_gene_list$table,aes(x=logCPM))+
    geom_histogram(aes(y=..density..),bins=50)+
    geom_density()+
    labs(title=analysis.name)
  
  # summary(all_gene_list$table$logCPM)
  #plot MA plot
  fig3 <- ggplot(lrt$table,aes(x=logCPM,y=logFC))+
    geom_point(alpha=0.3)+
    geom_point(data=gene_sig_up$table, aes(x=logCPM,y=logFC,color=color[2]),alpha=0.5)+
    geom_point(data=gene_sig_down$table, aes(x=logCPM,y=logFC,color=color[1]),alpha=0.5)+
    labs(color="",title=analysis.name)+
    scale_color_manual(labels=c("increased","decreased"),values=c(color[1],color[2]))+
    geom_hline(yintercept = c(0.6,-0.6),color="magenta",linetype="dashed")
    
  #volcano plot
  fig4 <- ggplot(lrt$table,aes(x=logFC,y=-log10(PValue)))+
    geom_point(alpha=0.3)+
    geom_point(data=gene_sig_up$table, aes(x=logFC,y=-log10(PValue),color=color[2]),alpha=0.5)+
    geom_point(data=gene_sig_down$table, aes(x=logFC,y=-log10(PValue),color=color[1]),alpha=0.5)+
    labs(color="",title=analysis.name)+
    scale_color_manual(labels=c("increased","decreased"),values=c(color[1],color[2]))+
    geom_vline(xintercept = c(0.6,-0.6),color="magenta",linetype="dashed")
  
  print(fig1)
  print(fig2)
  print(fig3)
  print(fig4)
  return(list(all_gene_list,gene_sigg_up,gene_sigg_down))
}

DE_test <- function(countabel.path,design,analysis.name,correlation.method="spearman",transform.method="log+1",fdr=0.1,logfc=0.6){
  #this function is to put all the above functions together.
  #input: as can be seen in above functions
  #output: filtered and normalized yy, gene_list that contains three toptag objects: all_gene_lsit, gene_sigg_up, gene_sigg_down)
  yy <- DGE_generater(countabel.path)
  yy <- normalize(yy,design)
  qc_bundle(yy,correlation.method,transform.method)
  gene_list<- GLM_fitting(yy,design,analysis.name,fdr,logfc)
  return(list(yy,gene_list))
}

```

##Analysis for total rna library

the countable is generated by iRNA_seq pipeline (featureCounts), use edgeR to analyze this countable

seems manually edgeR has much fewer significant genes.....

*FeatureCounts output by iRNA-seq:* [use the intron regions= longest isoform - genebank mRNA [to get more accurate intron annotations]  ]18938 genes for exon gtf files; 17285 genes for intron gtf files
1707 genes only in exon file but not intron gtf file: some are indeed single exon genes, but some are quite normal genes 

need to check on the gtf files

*take a deep look at total_intron MA plot*
*the outlier dots in MA plot:* they are all (1487N=0, 2139N=0, 1487T=c1, 2139T=c2; 1487N=c1, 2139N=c2, 1487T=0, 2139T=0)
one example of middle grey dots around significant genes: (although log2cpm=7.4, log2fc=2)
1487_basal 2139_basal 1487_treat 2139_treat 
  27.96552   29.10831  610.55523   21.31747 (cpm)
examples of highly significant genes: (SELE, CXCL3, ICAM1, VCAM1, CXCL1)
     1487_basal 2139_basal 1487_treat 2139_treat (cpm)
316    43.72075   52.46266  1773.2767  1593.4062
6526   23.23896   30.46219   666.6454   976.1001
133    33.08597   92.06351  1366.3932  1736.0231
1645   73.26179  124.55651   822.5025  1367.3208
6155  590.03315  752.07762  3006.3408  4329.2488

examples of sig down genes:
     1487_basal 2139_basal 1487_treat 2139_treat (cpm)
16241   26.783881    5.753969    0.0000000    0.0000000
8732   127.223436   96.802070   23.4475275   48.3396254
1901    55.931046   51.108785   14.2524187    6.3051685
8601   185.517766  142.156886   87.8132893   60.0492241
1486    72.867912   51.785723   12.8731524   14.1115677
10405  177.640154  150.280137   70.3425825   75.6620223
16441   44.902389   38.923909    0.4597554    9.3076297
13143    8.271493   19.292720    0.0000000    0.0000000

example of sig up genes:
     1487_basal 2139_basal 1487_treat 2139_treat (cpm)
17079   64.2025391   63.2936612  145.742475  140.515184
1132     6.6959703    0.0000000   32.642636   36.930273
7160     0.0000000    0.0000000   11.034131   13.811322
17275  241.4488126  281.2675532  622.508867  348.885992
7015   381.2764283  290.0677414  502.052942  579.475012
16381    2.7571643   24.3697519   43.676767   64.252670

##Real start
*Results*:
1. the distribution (logCPM) of exon and intron is totally different; and the polya_intron has more low counts genes compared to total_intron which may be because polya missed out a few nascent transcripts

2. if look at total_intron correlation matirx or the BCV plot, the spearman correaltion is relatively low and the variance is a little high....

3. the significant gene numbers:(FDR<0.1, abs(log2FC)>0.6)
        total_intron  total_exon  polya_intron  polya_exon
sig.up      275         115           348         172    
sig.donw    100         30            141         62

4. why so many in polya library???? (only FDR < 0.1)
snapshot of the last several genes of polya samples  {seems quite fine!!!!} polya_intron
       basal1     basal2    treat1      treat2
14135 26.4383296 19.1645962 36.745404  44.587148
15521 17.0569868 31.9409936 31.368028  51.227787
14947 49.4652618 52.9307894 60.943597  96.763598
16224 10.2341921  4.5629991 18.820817  17.075929
7448   0.8528493  0.0000000  7.169835   2.845988
14840 24.7326309 23.7275953 34.952946  48.381799
4136   7.6756441 10.0385980 16.132129  22.767905

last rows of polya_intron  (seems okay)
10548 14.4984388 10.9511978 25.990652 22.767905
2045  19.6155348  8.2133984 23.301964 39.843835
1513   0.0000000  3.6503993  1.792459 14.229941
15574  0.8528493  0.0000000  3.584917  5.691976
844    0.8528493  0.0000000  3.584917  5.691976
10982 23.0269322 19.1645962 34.056716 36.997846
8390  38.3782203 31.9409936 65.424744 41.741160
16798  1.7056987  0.9125998  5.377376  7.589302
4809   4.2642467  3.6503993 14.339670  7.589302
14178 17.0569868  4.5629991 25.094422 19.921917

the most logFCs: (logFC>5.5,polya_intron)
      basal1487  basal2139  treat1487  treat2139 (cpm)
6294 11.9398908 10.0385980 765.379885 469.588051
6352  1.7056987  1.8251996 105.755066  89.174296
481   0.8528493  2.7377995  95.896543  77.790344
4475  0.0000000  0.9125998  25.990652  24.665231
4639  0.0000000  0.0000000   6.273606  10.435290
6276  0.0000000  0.0000000   6.273606   3.794651

polya_intron sig down gene (last several rows in the table)
      basal1487  basal2139  treat1487  treat2139 (cpm)
13757  16.204137  10.951198   7.1698350   3.794651
14243  15.351288  11.863798   5.3773762   5.691976
10658  23.879782  31.028394  13.4434406  17.075929
12307   9.381343  11.863798   3.5849175   3.794651
15106  17.056987  20.077196   6.2736056  11.383953
16064  48.612412  33.766193  25.0944225  25.613894
5012   34.113974  40.154392  25.9906518  17.075929
1060    2.558548   2.737799   0.0000000   0.000000
11990  32.408275  42.892191  22.4057343  23.716568

snapshot of total_exon, sigup: (generall okay)
      basal1487  basal2139  treat1487  treat2139 (cpm)
2170    21.2710497 2.297226e+01   38.796002   37.271565
18069    1.1718608 3.929466e-01    7.274250    2.482911
7457     8.7712008 6.710319e+00   17.754103   14.757977
5566    10.4757257 6.166239e+00   19.562391   14.172122
2165    22.5139324 2.279090e+01   51.906092   30.882951
4128     1.7755467 3.506293e+00    8.671564    7.867202
8446     2.6633201 3.657426e+00    7.685225   10.378011
13290   17.4713797 8.493692e+00   28.809319   18.161518
6699    11.2214553 1.478084e+01   29.754561   21.397672
14631    3.6931372 7.012585e+00   14.630696   11.800802
1928     2.4857654 1.964733e+00    7.726322    6.807082
14213   10.3336819 8.342558e+00   25.973595   13.363083

snapshot of polya_exon,sigup: [seems weird: too low threshold to call significance?!]
      basal1487  basal2139  treat1487  treat2139 (cpm)
5092  8.215069e+00 7.032999e+00   10.529490 1.370280e+01
10157 3.286028e+00 3.146342e+00    5.360468 5.270307e+00
13671 4.576967e+00 4.195122e+00    7.466366 5.973015e+00
3647  2.523200e+00 3.084649e+00    4.147981 4.989224e+00
17909 1.290939e+00 2.282640e+00    3.382200 2.951372e+00
18742 1.584335e+00 1.110474e+00    2.488789 2.318935e+00
13141 2.581879e+00 1.295552e+00    3.892721 2.389206e+00
2087  2.171125e+00 1.295552e+00    3.509830 2.178394e+00
47    2.640558e+00 6.169297e-02    4.594687 7.027076e-02
15535 1.232260e+00 3.516499e+00    1.914453 5.340578e+00
1832  1.701693e+00 8.637016e-01    2.105898 2.178394e+00

5. the comparison between analysis: look at the old figures
New code is under debugging.......
(1). try different correlation methods
(2). remove 0 and 10~20% lowly expressed genes and compare (based on both logcpm and logFC)

Future Work:
1. logcpm and logFC correlation matrix
2. take a look at the gtf files of iRNA-seq
3. vennplot of the overlapped gene numbers
4. drug treatment analysis
5. is iRNA-seq worth for rapid response transcriptome characterization? (a bonus, but a large waste of sequencing reads; maybe gro-seq or longer time RNA-seq (more confident with reponsive genes) is worth trying)

```{r}
library(edgeR)
library(ggplot2)
library(RColorBrewer)
# create a design matrix for all the test
patient <- factor(c("1487","2139","1487","2139"))
tissue <- factor(c("N","N","T","T"))
design <- model.matrix(~patient+tissue)
rownames(design) <- c("1487N","2139N","1487T","2139T")

# the real start
countable.path=list(total.intron.table="/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_intron_counttable.txt",total.exon.table="/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_exon_counttable.txt",polya.intron.table="/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/polya_basal_treatment_intron_counttable.txt",polya.exon.table="/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/polya_basal_treatment_exon_counttable.txt")
total_intron_result <- DE_test(countable.path$total.intron.table,design,"total intron") #~10000 genes left after filtering
total_exon_result <- DE_test(countable.path$total.exon.table,design,"total exon") #~10000 genes left after filtering
polya_intron_result <- DE_test(countable.path$polya.intron.table,design,"polya intron") #~11000 genes left after filtering
polya_exon_result <- DE_test(countable.path$polya.exon.table,design,"polya exon")#~11000 genes left after filtering

# compare the reads count distribution of all samples
#polya intron counts distribution for each sample
color <- brewer.pal(n=4,name="Set1")
# total_intron sample
ggplot(data=cpm(total_intron_result[[1]]),aes(x=log2(basal1487),color=color[2]))+
  geom_density(bins=50,alpha=0.3)+
  geom_density(aes(x=log2(basal2139),color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=log2(treat1487),color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=log2(treat2139),color=color[2]),bins=50,alpha=0.3)+
  geom_density(data=cpm(total_exon_result[[1]]),aes(x=log2(basal1487),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(total_exon_result[[1]]),aes(x=log2(basal2139),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(total_exon_result[[1]]),aes(x=log2(treat1487),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(total_exon_result[[1]]),aes(x=log2(treat2139),color=color[1]),bins=50,alpha=0.3)+
  labs(title="total_RNA samples",x="log2cpm")+
  scale_color_discrete(labels=c("total_intron","total_exon","polya_intron","polya_exon"),name="Type")

#polya sample
ggplot(data=cpm(polya_intron_result[[1]]),aes(x=log2(basal1487),color=color[2]))+
  geom_density(bins=50,alpha=0.3)+
  geom_density(aes(x=log2(basal2139),color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=log2(treat1487),color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=log2(treat2139),color=color[2]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_exon_result[[1]]),aes(x=log2(basal1487),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_exon_result[[1]]),aes(x=log2(basal2139),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_exon_result[[1]]),aes(x=log2(treat1487),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_exon_result[[1]]),aes(x=log2(treat2139),color=color[1]),bins=50,alpha=0.3)+
  labs(title="polya_RNA samples",x="log2cpm")+
  scale_color_discrete(labels=c("intron","exon"),name="Type")

# incorportate the two libraries together
ggplot(data=cpm(total_intron_result[[1]]),aes(x=log2(basal1487),color=color[2]))+
  geom_density(bins=50,alpha=0.3)+
  geom_density(aes(x=log2(basal2139),color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=log2(treat1487),color=color[2]),bins=50,alpha=0.3)+
  geom_density(aes(x=log2(treat2139),color=color[2]),bins=50,alpha=0.3)+
  geom_density(data=cpm(total_exon_result[[1]]),aes(x=log2(basal1487),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(total_exon_result[[1]]),aes(x=log2(basal2139),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(total_exon_result[[1]]),aes(x=log2(treat1487),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(total_exon_result[[1]]),aes(x=log2(treat2139),color=color[1]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_intron_result[[1]]),aes(x=log2(basal1487),color=color[3]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_intron_result[[1]]),aes(x=log2(basal2139),color=color[3]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_intron_result[[1]]),aes(x=log2(treat1487),color=color[3]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_intron_result[[1]]),aes(x=log2(treat2139),color=color[3]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_exon_result[[1]]),aes(x=log2(basal1487),color=color[4]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_exon_result[[1]]),aes(x=log2(basal2139),color=color[4]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_exon_result[[1]]),aes(x=log2(treat1487),color=color[4]),bins=50,alpha=0.3)+
  geom_density(data=cpm(polya_exon_result[[1]]),aes(x=log2(treat2139),color=color[4]),bins=50,alpha=0.3)+
  labs(title="total vs polya RNA samples",x="log2cpm")+
  scale_color_discrete(labels=c("total_intron","polya_intron","polya_exon","total_exon"),name="Type")hjupofdasjkl
  



```

1. tried to compare all the 8 groups at one round, but only get 3000 genes, even without filtering 10% lowly expressed genes; so can only do pairwise comparisons;
tried with the threshold 10%, thus still have ~6000 genes to plot 


```{r} 
#temporarily storing things
# rname<- rownames(total_exon_result[[2]][[2]])
# cpm(total_exon_result[[1]])[rname,]
library(corrplot)

#test if the libraries are comparable between each other [group: total_intron_basal, total_intron_treat, total_exon_basal..... 8 groups in total]; get the mean of teh two replicates,remove the 0 expression genes and 20% lowly expressed genes and use the overlap gene of the 8 groups
#start from the 4 yy, the preprocess step
total_intron_basal <- 0.5*(cpm(total_intron_result[[1]])[,"basal1487"]+cpm(total_intron_result[[1]])[,"basal2139"])
total_intron_basal <- log2(total_intron_basal[total_intron_basal>0])
total_intron_basal <- total_intron_basal[1:round(length(total_intron_basal)*0.8)]

total_intron_treat <- 0.5*(cpm(total_intron_result[[1]])[,"treat1487"]+cpm(total_intron_result[[1]])[,"treat2139"])
total_intron_treat <- log2(total_intron_treat[total_intron_treat>0])
total_intron_treat <- total_intron_treat[1:round(length(total_intron_treat)*0.8)]

total_exon_basal <- 0.5*(cpm(total_exon_result[[1]])[,"basal1487"]+cpm(total_exon_result[[1]])[,"basal2139"])
total_exon_basal <- log2(total_exon_basal[total_exon_basal>0])
total_exon_basal <- total_exon_basal[1:round(length(total_exon_basal)*0.8)]

total_exon_treat <- 0.5*(cpm(total_exon_result[[1]])[,"treat1487"]+cpm(total_exon_result[[1]])[,"treat2139"])
total_exon_treat <- log2(total_exon_treat[total_exon_treat>0])
total_exon_treat <- total_exon_treat[1:round(length(total_exon_treat)*0.8)]

#polya library
polya_intron_basal <- 0.5*(cpm(polya_intron_result[[1]])[,"basal1487"]+cpm(polya_intron_result[[1]])[,"basal2139"])
polya_intron_basal <- log2(polya_intron_basal[polya_intron_basal>0])
polya_intron_basal <- polya_intron_basal[1:round(length(polya_intron_basal)*0.8)]

polya_intron_treat <- 0.5*(cpm(polya_intron_result[[1]])[,"treat1487"]+cpm(polya_intron_result[[1]])[,"treat2139"])
polya_intron_treat <- log2(polya_intron_treat[polya_intron_treat>0])
polya_intron_treat <- polya_intron_treat[1:round(length(polya_intron_treat)*0.8)]

polya_exon_basal <- 0.5*(cpm(polya_exon_result[[1]])[,"basal1487"]+cpm(polya_exon_result[[1]])[,"basal2139"])
polya_exon_basal <- log2(polya_exon_basal[polya_exon_basal>0])
polya_exon_basal <- polya_exon_basal[1:round(length(polya_exon_basal)*0.8)]

polya_exon_treat <- 0.5*(cpm(polya_exon_result[[1]])[,"treat1487"]+cpm(polya_exon_result[[1]])[,"treat2139"])
polya_exon_treat <- log2(polya_exon_treat[polya_exon_treat>0])
polya_exon_treat <- polya_exon_treat[1:round(length(polya_exon_treat)*0.8)]

########################################
# overlap the gene list,,, the low correlation is due to the rownames are not corresponding to each one !!!!!!!!!!!!!! this could affect the correlation matrix plot above!
comparison.group <- list(total_intron_basal,total_intron_treat,total_exon_basal,total_exon_treat,polya_intron_basal,polya_intron_treat,polya_exon_basal,polya_exon_treat)
M <- matrix(,nrow=8,ncol=8) #preallocate a matrix (NA) for correlation matrix
rownames(M) <- c("total_intron_basal","total_intron_treat","total_exon_basal","total_exon_treat","polya_intron_basal","polya_intron_treat","polya_exon_basal","polya_exon_treat")
colnames(M) <- c("total_intron_basal","total_intron_treat","total_exon_basal","total_exon_treat","polya_intron_basal","polya_intron_treat","polya_exon_basal","polya_exon_treat")
filtered.genes <- list(total_intron_result[[1]]$genes,total_intron_result[[1]]$genes,total_exon_result[[1]]$genes,total_exon_result[[1]]$genes,polya_intron_result[[1]]$genes,polya_intron_result[[1]]$genes,polya_exon_result[[1]]$genes,polya_exon_result[[1]]$genes)
correlation.calculator<- function(comparison.group,filtered.genes,method="pearson") { 
  #this function is to make the correlation matrix plot and output the correlation matrix
  #input: .....
  #output: the matrix to store each pairwise correlation coefficients
  for (i in 1:length(comparison.group)){
    for (j in 1:length(comparison.group)){
          if (i<=j){
              ###gene names is factor, and the number is uniq for each group so first thing is to coerce it into strings and then compare!!!!!
              idx1 <- filtered.genes[[i]][names(comparison.group[[i]]),] %in% filtered.genes[[j]][names(comparison.group[[j]]),] #idx1 is for the first group (logical)
              gene_name <- filtered.genes[[i]][names(comparison.group[[i]]),]
              gene_name <- gene_name[idx1]
              print(length(gene_name))
              #use genename as an ID to subset both variables; firstly convert the gene name to level id for the two
              searchID1=vector("character",length(gene_name))
              searchID2=vector("character",length(gene_name))
              for (ii in 1:length(gene_name)){
                    idxx1 <- filtered.genes[[i]]==as.character(gene_name[ii])
                    searchID1[ii]=rownames(filtered.genes[[i]])[idxx1]
                    
                    idxx2 <- filtered.genes[[j]]==as.character(gene_name[ii])
                    searchID2[ii]=rownames(filtered.genes[[j]])[idxx2]
                    
              }
              M[i,j]<- cor(comparison.group[[i]][searchID1],comparison.group[[j]][searchID2],method=method)
          }
    }
  }
  M1 <- M
  for (i in 1:8){
        for (j in 1:8){
              if (i <j){
                    M1[j,i]=M1[i,j]
              }
        }
  }
  corrplot.mixed(M1,tl.cex=0.6)
  return(M1)
}
M1 <- correlation.calculator(comparison.group,filtered.genes) # so I have already checked that the pearson and spearman correlation have similar results.
# M2 <- correlation.calculator(comparison.group,filtered.genes,method="spearman")


########################### 
#this is an instance of the above loop!
# ###gene names is factor, and the number is uniq for each group so first thing is to coerce it into strings and then compare!!!!!
# idx1 <- total_exon_result[[1]]$genes[names(total_exon_basal),] %in% polya_exon_result[[1]]$genes[names(polya_exon_basal),] #idx1 is for the first group (logical)
# gene_name <- total_exon_result[[1]]$genes[names(total_exon_basal),]
# gene_name <- gene_name[idx1]
# print(length(gene_name))
# #use genename as an ID to subset both variables; firstly convert the gene name to level id for the two
# searchID1=vector("character",length(gene_name))
# searchID2=vector("character",length(gene_name))
# for (ii in 1:length(gene_name)){
#       idxx1 <- total_exon_result[[1]]$genes==as.character(gene_name[ii])
#       searchID1[ii]=rownames(total_exon_result[[1]]$genes)[idxx1]
#       
#       idxx2 <- polya_exon_result[[1]]$genes==as.character(gene_name[ii])
#       searchID2[ii]=rownames(polya_exon_result[[1]]$genes)[idxx2]
#       
# }
# cor(total_exon_basal[searchID1],polya_exon_basal[searchID2])
##########################
# the correlation based on logcpm
logFC_mat <- matrix(,nrow=4,ncol=4)
rownames(logFC_mat) <- c("total_intron","total_exon","polya_intron","polya_exon")
colnames(logFC_mat) <- c("total_intron","total_exon","polya_intron","polya_exon")
analysis.list <- list(total_intron_result[[2]][[1]],total_exon_result[[2]][[1]],polya_intron_result[[2]][[1]],polya_exon_result[[2]][[1]])
for (ii in 1:4){
  for (jj in 1:4){
  gene.number1<- nrow(analysis.list[[ii]])
  o1 <- order(analysis.list[[ii]]$table$logCPM,decreasing = T)
  temp1 <- analysis.list[[ii]]$table[o1,] # order the list based on logcpm
  temp1 <- temp1[1:round(gene.number1*0.9),]
  
  gene.number2<- nrow(analysis.list[[jj]])
  o2 <- order(analysis.list[[jj]]$table$logCPM,decreasing = T)
  temp2 <- analysis.list[[jj]]$table[o2,] # order the list based on logcpm
  temp2 <- temp2[1:round(gene.number2*0.9),]
  idx1 <- temp1$genes %in% temp2$genes
  gene.names <- temp1$genes[idx1]
  idx2 <- vector("numeric",length(gene.names))
  for (i in 1:length(gene.names)){
    idx2[i]=which(temp2$genes==as.character(gene.names[i]))
  }
  logFC_mat[ii,jj] <- cor(temp1[idx1,]$logFC,temp2[idx2,]$logFC)
  }
}
corrplot.mixed(logFC_mat,order="hclust")


###############################

#filter out 10% logcpm genes
gene.number1<- nrow(total_exon_result[[2]][[1]])
o1 <- order(total_exon_result[[2]][[1]]$table$logCPM,decreasing = T)
temp1 <- total_exon_result[[2]][[1]]$table[o1,] # order the list based on logcpm
temp1 <- temp1[1:round(gene.number1*1),]

gene.number2<- nrow(polya_exon_result[[2]][[1]])
o2 <- order(polya_exon_result[[2]][[1]]$table$logCPM,decreasing = T)
temp2 <- polya_exon_result[[2]][[1]]$table[o2,] # order the list based on logcpm
temp2 <- temp2[1:round(gene.number2*1),]


idx1 <- temp1$genes %in% temp2$genes
gene.names <- temp1$genes[idx1]
idx2 <- vector("numeric",length(gene.names))
for (i in 1:length(gene.names)){
  idx2[i]=which(temp2$genes==as.character(gene.names[i]))
}
cor(temp1[idx1,]$logFC,temp2[idx2,]$logFC)

##############################

overlap.names <- names(polya_intron_basal)[names(polya_intron_basal) %in% names(total_intron_basal)]
mat=cor(polya_intron_basal[overlap.names],total_intron_basal[overlap.names],method="spearman")

comparison.group <- lapply(comparison.group, names) 
overlap.genes <- Reduce(intersect,comparison.group)

```

# comparsisons between TNFa and TNFa+TPCA-1
the result is pretty weird: only find two significant genes from intron analysis; no significant genes from exon analysis??????????
if I use whole iRNA-seq pipeline; supposed to find hundreds of significant genes

```{r}
# create a design matrix for all the test
patient1 <- factor(c("1487","2139","1487","2139"))
tissue1 <- factor(c("TPCA","TPCA","Treat","Treat"))
inhibit.design <- model.matrix(~patient1+tissue1)
rownames(inhibit.design) <- c("1487TPCA","2139TPCA","1487Treat","2139Treat")

# the real start
inhibit.path=list(intron.table="/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/drug_treatment/tpca_treatment_intron_countable.txt",exon.table="/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/drug_treatment/tpca_treatment_exon_countable.txt")
inhibit.intron.result <- DE_test(inhibit.path$intron.table, design, "TPCA vs TNFa intron")
inhibit.exon.result <- DE_test(inhibit.path$exon.table, design, "TPCA vs TNFa exon")




```

