---
title: "iRNA_seq_test"
output: html_document
---

#start here
**conclusion**: actually the data seems okay; although exon looks similar to intron; need to check on
ploy-A data; or do it manually; then can tell the difference.

```{r}
intron <- read.table("~/Desktop/Dale's/iRNA_seq_results/basal_treatment_intron.txt",header=T,sep='\t')
exon <- read.table("~/Desktop/Dale's/iRNA_seq_results/basal_treatment_exon.txt",header=T,sep='\t')
names(intron)[14:17] <- c("1487_basal","2139_basal","1487_treat","2139_treat")
names(exon)[14:17] <- c("1487_basal","2139_basal","1487_treat","2139_treat")

```

look at the p_value distribution; since edgeR will give some bad quality genes (maybe low_counts or something, can check on the documentation) p_value=1; so if not remove the p_value=1 the another mode close to 1

```{r}
hist(intron[intron$Pval!=1,]$Pval,breaks=100)
hist(exon[exon$Pval!=1,]$Pval,breaks=100)

```
```{r}
hist(intron$Padj,breaks=100)
sum(intron$Padj<.1)
sum(exon$Padj<.1)
intron_sig <- intron[intron$Padj<0.1,]
exon_sig <- exon[exon$Padj<0.1,]
print("sigup_intron");sum(intron_sig$Log2_FC>0)
print("sigdown_intron");sum(intron_sig$Log2_FC<0)
print("sigup_exon");sum(exon_sig$Log2_FC>0)
print("sigdown_exon");sum(exon_sig$Log2_FC<0)

hist(exon_sig$Log2_FC,breaks=100)
hist(intron_sig$Log2_FC,breaks=100)
```

MA_plot:
```{r MA_plot}
par(pch=19)
plot(intron$Log2_FC~intron$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3))
abline(h=0,col="magenta")
intron_sig_up <- intron_sig[intron_sig$Log2_FC>0,]
intron_sig_down<- intron_sig[intron_sig$Log2_FC<0,]
points(intron_sig_up$Log2_FC~intron_sig_up$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(intron_sig_down$Log2_FC~intron_sig_down$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)

#exon
plot(exon$Log2_FC~exon$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3))
abline(h=0,col="magenta")
exon_sig_up <- exon_sig[exon_sig$Log2_FC>0,]
exon_sig_down<- exon_sig[exon_sig$Log2_FC<0,]
points(exon_sig_up$Log2_FC~exon_sig_up$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(exon_sig_down$Log2_FC~exon_sig_down$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
```


#poly_rna_seq
```{r}
polya_intron <- read.table("~/Desktop/Dale's/iRNA_seq_results/polya_basal_treatment_intron.txt",header=T,sep='\t')
polya_exon <- read.table("~/Desktop/Dale's/iRNA_seq_results/polya_basal_treatment_exon.txt",header=T,sep='\t')
names(intron)[14:17] <- c("1487_basal","2139_basal","1487_treat","2139_treat")
names(exon)[14:17] <- c("1487_basal","2139_basal","1487_treat","2139_treat")
#pvalue_distribution
hist(polya_intron[polya_intron$Pval!=1,]$Pval,breaks=100)
hist(polya_exon[polya_exon$Pval!=1,]$Pval,breaks=100)

```

log2_FC distribution
```{r}
hist(polya_intron$Padj,breaks=100)
sum(polya_intron$Padj<.1)
sum(polya_exon$Padj<.1)
polya_intron_sig <- polya_intron[polya_intron$Padj<0.1,]
polya_exon_sig <- polya_exon[polya_exon$Padj<0.1,]
print("sigup_intron");sum(polya_intron_sig$Log2_FC>0)
print("sigdown_intron");sum(polya_intron_sig$Log2_FC<0)
print("sigup_exon");sum(polya_exon_sig$Log2_FC>0)
print("sigdown_exon");sum(polya_exon_sig$Log2_FC<0)

hist(polya_exon_sig$Log2_FC,breaks=100)
hist(polya_intron_sig$Log2_FC,breaks=100)

print("siggup_intron");sum(polya_intron_sig$Log2_FC>1)
print("siggdown_intron");sum(polya_intron_sig$Log2_FC< -1)
print("siggup_exon");sum(polya_exon_sig$Log2_FC>1)
print("siggdown_exon");sum(polya_exon_sig$Log2_FC< -1)
```
MA_plot:
```{r}
par(pch=19)
plot(polya_intron$Log2_FC~polya_intron$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3))
abline(h=0,col="magenta")
polya_intron_sig_up <- polya_intron_sig[polya_intron_sig$Log2_FC>0,]
polya_intron_sig_down<- polya_intron_sig[polya_intron_sig$Log2_FC<0,]
points(polya_intron_sig_up$Log2_FC~polya_intron_sig_up$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(polya_intron_sig_down$Log2_FC~polya_intron_sig_down$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
abline(h=0.6,col="magenta",lty=2)
abline(h=-0.6,col="magenta",lty=2)
abline(h=1,col="magenta")
abline(h=-1,col="magenta")
#exon
plot(polya_exon$Log2_FC~polya_exon$logCPM,cex=0.7,col=rgb(0,0,0,alpha=0.3))
abline(h=0,col="magenta")
polya_exon_sig_up <- polya_exon_sig[polya_exon_sig$Log2_FC>0,]
polya_exon_sig_down<- polya_exon_sig[polya_exon_sig$Log2_FC<0,]
points(polya_exon_sig_up$Log2_FC~polya_exon_sig_up$logCPM,col=rgb(0.7,0,0,alpha=0.3),cex=0.7)
points(polya_exon_sig_down$Log2_FC~polya_exon_sig_down$logCPM,col=rgb(0,0,0.7,alpha=0.3),cex=0.7)
abline(h=0.6,col="magenta",lty=2)
abline(h=-0.6,col="magenta",lty=2)
abline(h=1,col="magenta")
abline(h=-1,col="magenta")

```
