library(edgeR)
## read a countable and create a DGElist out from it
DGE_generater <- function(countable_path){
  #input: path of a gene counttable generated by iRNA-seq; output: one DGEList object
  # counttable_path <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_intron_counttable.txt"
  # counttable_path2 <- "/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_exon_counttable.txt"
  intron_table <- read.table(countable_path,sep="\t",header=T)
  colnames(intron_table)[9:12] <- c('1487_basal',"2139_basal","1487_treat","2139_treat")
  yy <- DGEList(counts=intron_table[9:12],genes=intron_table$Symbol,group=factor(c("N","N","T","T")))
  return(yy)
}
yy <- DGE_generater("/home/mdwilson/hpf_trial/liangxi/projects/rela/bam/rna_seq/iRNA_seq_results/basal_treatment_intron_counttable.txt")

## preprocess the DGElist and print some qc figures
normalize <- function(yy,design){
  #input:DGElist object and a design object; output: a normalized DGElist object
  #reordering and filtering the yy; calculate the lib.size and the dispersions;plot the default "PCA" and the BCV
  o <- order(rowSums(yy$counts),decreasing = T)
  yy <- yy[o,]
  idx <- rowSums(cpm(yy)>2)>=2  # ~10000 genes left
  yy <- yy[idx,]
  yy$samples$lib.size <- colSums(yy$counts)
  yy <- calcNormFactors(yy)
  # yy$samples  # to show the lib.size and the norm.factoers
  plotMDS(yy,cex=0.7)
  yy <- estimateGLMCommonDisp(yy,design)
  yy <- estimateGLMTrendedDisp(yy,design)
  yy <- estimateGLMTagwiseDisp(yy,design)
  plotBCV(yy,cex=0.4)
  return(yy)
}
patient <- factor(c("1487","2139","1487","2139"))
tissue <- factor(c("N","N","T","T"))
design <- model.matrix(~patient+tissue)
rownames(design) <- c("1487N","2139N","1487T","2139T")
yy <- normalize(yy,design)

## a bunch of qc_plot to know about the intrinsic data structure (correlation matrix, PCA, correlation heatmap)
qc_bundle <- function(yy, correlation.method="spearman",transform.method="log+1"){
  #plot correlation matrix, PCA, correlation heatmap between samples
  #input: a normalized DGElist generated by normalize function, a character to say the correlation method used in correlatin matirx, a character to show how to transform the data
  library(corrplot)
  library(ggfortify)
  transformed.yy <- data.frame(cpm(yy))
  if (transform.method=="log+1"){
    transformed.yy <- log(transformed.yy+1) 
  }else if (transform.method=="log"){
    transformed.yy <- log(transformed.yy) 
  }else {
    return(1)
  }
  transformed.yy <- as.data.frame(t(transformed.yy)) 
  #correlatin matirx
  M <- cor(transformed.yy,method=correlation.method)
  fig1<- corrplot.mixed(M,order="hclust")
  #PCA
  transformed.yy.pca <- prcomp(transformed.yy,
                center=T,
                scale.=T)
  #print(transformed.yy.pca)
  #summary(transformed.yy.pca)
  y2<- cbind(transformed.yy2,treatment=c("1487_basal" ,"2139_basal" ,"1487_treat", "2139_treat")) #with group labels
  plot(transformed.yy.pca, type="l")
  fig2 <- autoplot(transformed.yy2.pca,label=T,data=y2,colour="treatment")
  #heatmap
  sample_dist <- dist(1-cor(transformed.yy)) #perform euclidean distance on the normalized and transformed data, pearson correlation; 1-pearson correlation = distance
  hc <- hclust(sample_dist,method="average")
  fig3 <- heatmap.2(cor(transformed.yy),Rowv=as.dendrogram(hc),symm=T,trace="none")
  print(fig1)
  print(fig2)
  print(fig3)
}
qc_bundle(yy,"spearman","log+1")
